validation-test/stdlib/Dictionary.swift

```diff
@@ -83,6 +83,28 @@ DictionaryTestSuite.test("Index.Hashable") {
   expectNotNil(e[d.startIndex])
 }
 
+DictionaryTestSuite.test("Hashable") {
+    let d1: Dictionary<Int, String> = [1: "meow", 2: "meow", 3: "meow"]
+    let d2: Dictionary<Int, String> = [1: "meow", 2: "meow", 3: "mooo"]
+    let d3: Dictionary<Int, String> = [1: "meow", 2: "meow", 4: "meow"]
+    let d4: Dictionary<Int, String> = [1: "meow", 2: "meow", 4: "mooo"]
+    checkHashable([d1, d2, d3, d4], equalityOracle: { $0 == $1 })
+
+    let dd1: Dictionary<Int, Dictionary<Int, String>> = [1: [2: "meow"]]
+    let dd2: Dictionary<Int, Dictionary<Int, String>> = [2: [1: "meow"]]
+    let dd3: Dictionary<Int, Dictionary<Int, String>> = [2: [2: "meow"]]
+    let dd4: Dictionary<Int, Dictionary<Int, String>> = [1: [1: "meow"]]
+    let dd5: Dictionary<Int, Dictionary<Int, String>> = [2: [2: "mooo"]]
+    let dd6: Dictionary<Int, Dictionary<Int, String>> = [2: [:]]
+    let dd7: Dictionary<Int, Dictionary<Int, String>> = [:]
+    checkHashable([dd1, dd2, dd3, dd4, dd5, dd6, dd7], equalityOracle: { $0 == $1 })
+
+    // d5 is equal to d4, but will probably be iterated differently
+    let d5: Dictionary<Int, String> = [1: "meow", 4: "mooo", 2: "meow"]
```

Comment:
A more reliable way to get different ordering for the same values is to use `reserveCapacity`; each capacity gets a potentially different permutation. (This works too, but only as long as some of the keys collide.)

Reply:
Hi @lorentey ,
I'm not 100% certain that I understood you correctly, but in my latest commit, I tried instantiating two different mutable Dictionaries and reservedCapacity for 3 elements for each of them. Then I assigned values for the keys in different order and tested this with a version of the Dictionary-hashing, that (intentionally wrongfully) imposes the traversal order on the generated hash value. 
But the tests still succeed although I would expect them to fail.
Have I perhaps understood you wrong?